src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RijndaelManagedTransform.Browser.cs

```diff
@@ -0,0 +1,1282 @@
+    {
+        public const int BlockSizeBytes = 16; // 128 bits
+        private const int BlockSizeInts = BlockSizeBytes / 4;
+
+        private readonly bool _encrypting;
+
+        private int[] _encryptKeyExpansion;
+        private int[] _decryptKeyExpansion;
+
+        private readonly int _Nr;
+        private readonly int _Nk;
+
+        private readonly int[] _encryptindex;
+        private readonly int[] _decryptindex;
+
+        private int[] _IV;
+        private int[] _lastBlockBuffer;
+        private byte[]? _depadBuffer;
+
+        public RijndaelManagedTransform(ReadOnlySpan<byte> key,
+                                        ReadOnlySpan<byte> iv,
+                                        bool encrypting)
+        {
+            _encrypting = encrypting;
+            _Nr = GetNumberOfRounds(key);
+            _Nk = key.Length / 4;
+
+            // Precompute the modulus operations: these are performance killers when called frequently
+            _encryptindex = new int[BlockSizeInts * 3];
+            _decryptindex = new int[BlockSizeInts * 3];
+
+            Span<int> encryptindex1 = _encryptindex.AsSpan(0, BlockSizeInts);
+            Span<int> encryptindex2 = _encryptindex.AsSpan(BlockSizeInts, BlockSizeInts);
+            Span<int> encryptindex3 = _encryptindex.AsSpan(BlockSizeInts * 2, BlockSizeInts);
+
+            Span<int> decryptindex1 = _decryptindex.AsSpan(0, BlockSizeInts);
+            Span<int> decryptindex2 = _decryptindex.AsSpan(BlockSizeInts, BlockSizeInts);
+            Span<int> decryptindex3 = _decryptindex.AsSpan(BlockSizeInts * 2, BlockSizeInts);
+
+            for (int j = 0; j < BlockSizeInts; j++)
+            {
+                encryptindex1[j] = (j + 1) % BlockSizeInts;
+                encryptindex2[j] = (j + 2) % BlockSizeInts;
+                encryptindex3[j] = (j + 3) % BlockSizeInts;
+                decryptindex1[j] = (j - 1 + BlockSizeInts) % BlockSizeInts;
+                decryptindex2[j] = (j - 2 + BlockSizeInts) % BlockSizeInts;
+                decryptindex3[j] = (j - 3 + BlockSizeInts) % BlockSizeInts;
+            }
+
+            // we only support CBC mode
+            if (iv.Length / 4 != BlockSizeInts)
+                throw new CryptographicException(SR.Cryptography_InvalidIVSize);
+
+            _IV = new int[BlockSizeInts];
+            int index = 0;
+            for (int i = 0; i < BlockSizeInts; ++i)
+            {
+                int i0 = iv[index++];
+                int i1 = iv[index++];
+                int i2 = iv[index++];
+                int i3 = iv[index++];
+                _IV[i] = i3 << 24 | i2 << 16 | i1 << 8 | i0;
+            }
+
+            GenerateKeyExpansion(key);
+
+            _lastBlockBuffer = _IV.AsSpan().ToArray();
+        }
+
+        public void Dispose()
+        {
+            // We need to always zeroize the following fields because they contain sensitive data
+            if (_IV != null)
+            {
+                Array.Clear(_IV);
```

Comment:
I think in the spirit of "if we are going to bother zeroing things, might as well do it the """right""" way" using `CryptographicOperations.ZeroMemory`.

Reply:
`CryptographicOperations.ZeroMemory` only works on `Span<byte>`. These are `int[]`.

So I could do one of the following:

1. Unsafe cast the `int[]` to a `Span<byte>`.
2. Add a new overload of `CryptographicOperations.ZeroMemory` for `Span<int>`.

I wasn't sure if either of these were "worth it". Thoughts?

Reply:
Ah, no, you're right. I think it's probably fine to leave alone for now.